src/js/components/Chart/Chart.js

```diff
@@ -111,125 +116,138 @@ const renderArea = (
   );
 };
 
-class Chart extends Component {
-  static defaultProps = {
-    color: 'accent-1',
-    overflow: false,
-    size: { width: 'medium', height: 'small' },
-    thickness: 'medium',
-    type: 'bar',
-  };
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const { bounds, values } = nextProps;
-    const { bounds: stateBounds, values: stateValues } = prevState;
-    if (!stateValues || values !== stateValues || bounds !== stateBounds) {
-      const nextValues = normalizeValues(values);
-      const nextBounds = normalizeBounds(bounds, nextValues);
-      return { bounds: nextBounds, values: nextValues };
-    }
-    return null;
-  }
+const Chart = props => {
+  const {
+    color,
+    onClick,
+    onHover,
+    overflow,
+    round,
+    size,
+    theme,
+    thickness,
+    type,
+    values,
+    bounds,
+    ...rest
+  } = props;
 
-  containerRef = createRef();
+  const containerRef = useRef(null);
 
-  state = { containerWidth: 0, containerHeight: 0 };
+  const [state, setState] = useState({
+    containerWidth: 0,
+    containerHeight: 0,
+  });
+  const {
+    containerWidth,
+    containerHeight,
+    values: valuesState,
+    bounds: boundsState,
+  } = state;
 
-  componentDidMount() {
-    window.addEventListener('resize', this.onResize);
-    this.onResize();
-  }
+  const nextValues = normalizeValues(values);
+  const nextBounds = normalizeBounds(bounds, nextValues);
 
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.onResize);
+  if (
+    !valuesState ||
+    (!areNormalizedValuesEquals(values, valuesState) &&
+      !areNormalizedValuesEquals(valuesState, nextValues)) ||
+    (!areNormalizedBoundsEquals(bounds, boundsState) &&
+      !areNormalizedBoundsEquals(boundsState, nextBounds))
+  ) {
+    setState({ ...state, bounds: nextBounds, values: nextValues });
   }
 
-  onResize = () => {
-    const containerNode = this.containerRef.current;
+  const onResize = () => {
+    const containerNode = containerRef.current;
     if (containerNode) {
       const { parentNode } = containerNode;
       if (parentNode) {
         const rect = parentNode.getBoundingClientRect();
-        this.setState({
+        setState({
```

Comment:
Can we do this instead?
```js
setState(prevState => ({
  ...prevState,
  containerWidth: rect.width,
  containerHeight: rect.height,
}));
```

Otherwise, the spreaded state (`...state`) will always be the initial state values, not the latest.

This is kinda hard to explain for me. But I'll try.

`onResize` will be redefined every time the component is rendered.
On the first render, we define `onResize`, then our `useEffect(() => { ... },[])` runs and attaches the listener to the `window`.
Then our state/props change and the component rerenders. We define our `onResize` again, but we never use it, since it is not listed in our `useEffect` dependencies array.

So, whenever our `onResize` runs later on, it has stale `state` values.
So the simple fix is to use the functional of `setState` here.
If we were using state or props values to conditionally update the state, then it'd become a bit messier.

A true refactor according to [this detailed article](https://overreacted.io/a-complete-guide-to-useeffect) would be
```js
const onResize = useCallback(() => { ... }, []);
useEffect(() => { /* attach listener here */ }, [onResize]); 
```
Now if we every use state/props in `onResize` function, we would add it as the dependency in `useCallback` and our effect will rerun.

I think there should be `eslint-plugin-react-hooks` in the library. `exhaustive-deps` rule will help to catch unnoticed bugs like these.